front/itilfollowup.form.php

```diff
@@ -80,7 +84,45 @@
    Event::log($fup->getField('items_id'), strtolower($_POST['itemtype']), 4, "tracking",
               //TRANS: %s is the user login
               sprintf(__('%s purges a followup'), $_SESSION["glpiname"]));
-   Html::redirect($track->getFormURLWithID($fup->getField('items_id')));
+   $redirect = $track->getFormURLWithID($fup->getField('items_id'));
+}
+
+if ($handled) {
+   if (isset($_POST['kb_linked_id'])) {
+      //if followup should be linked to selected KB entry
+      $params = [
+         'knowbaseitems_id' => $_POST['kb_linked_id'],
+         'itemtype'         => $track->getType(),
+         'items_id'         => $track->getID()
+      ];
+      $existing = $DB->request(
+         'glpi_knowbaseitems_items',
+         $params
+      );
+      if ($existing->numrows() == 0) {
+         $kb_item_item = new KnowbaseItem_Item();
+         $kb_item_item->add($params);
+      }
+   }
+
+   if ($track->can($_POST["items_id"], READ)) {
+      $toadd = '';
+      // Copy followup to KB redirect to KB
+      if (isset($_POST['_fw_to_kb']) && $_POST['_fw_to_kb']) {
```

Comment:
For what I can see in current code; the used abbreviation for follow-up is "fup", not "fw"
```suggestion
      if (isset($_POST['_fup_to_kb']) && $_POST['_fup_to_kb']) {
```
This should of course be changed everywhere